src/main/java/net/neoforged/neoforge/common/extensions/ILivingEntityExtension.java

```diff
@@ -64,4 +66,15 @@ default boolean canDrownInFluidType(FluidType type) {
     default boolean moveInFluid(FluidState state, Vec3 movementVector, double gravity) {
         return state.move(self(), movementVector, gravity);
     }
+
+    /**
+     * Executes in {@link LivingEntity#hurt(DamageSource, float)} after all damage and
+     * effects have applied. Overriding this method is preferred over overriding the
+     * hurt method in custom entities where special behavior is desired after vanilla
```

Comment:
So in my case where I override hurt in my entity to cap the max damage to never more than 3, I would need to stop overriding this in my common code somehow and override onDamageTaken instead by mixin into my common entity class and overriding this method instead?
https://github.com/TelepathicGrunt/Bumblezone/blob/2e853574f4ca68e1a644de4f6c8daef13108794f/common/src/main/java/com/telepathicgrunt/the_bumblezone/entities/living/CosmicCrystalEntity.java#L1406

What is the issues that can happen if I refuse to stop overriding hurt?

Reply:
I would suggest calling `super.hurt` at the end of your override and including in your override only the parts that differ.  that said, looking at your override, you may want to just include the logic for adding to the damageContainer stack, firing the first event, and then popping the stack.  Whichever is cleaner, but both are options.

overriding `onDamageTaken` would be too late in the damage sequence to be of use for limiting the damage.  That callback is there for behavior that happens at the time of damage received but doesn't influence the damage sequence, eg. emitting particles on the entity based on the damage source and value as a custom mob feature.